engine/src/main/scala/cromwell/webservice/PartialWorkflowSources.scala

```diff
@@ -96,30 +106,54 @@ object PartialWorkflowSources {
       case _ => s"Incomplete workflow submission: $partialSource".invalidNel
     }
 
+    def validateWorkflowType(partialSource: PartialWorkflowSources, version: Int): ErrorOr[Option[WorkflowType]] = {
+      (partialSource.workflowType, version) match {
+        case (Some(src), _) => Option(src).validNel
+        case (None, 1) => None.validNel
+        case (None, _) => s"Workflow type is mandatory for v$version".invalidNel
+      }
+    }
+
+    def validateWorkflowTypeVersion(partialSource: PartialWorkflowSources,
+                                    version: Int): ErrorOr[Option[WorkflowTypeVersion]] = {
+      (partialSource.workflowTypeVersion, version) match {
+        case (Some(src), _) => Option(src).validNel
+        case (None, 1) => None.validNel
+        case (None, _) => s"Workflow type version is mandatory for v$version".invalidNel
+      }
+    }
+
     partialSources match {
       case Valid(partialSource) =>
-        (validateWorkflowSource(partialSource) |@| validateInputs(partialSource) |@| validateOptions(partialSource.workflowOptions)) map {
-          case (wfSource, wfInputs, wfOptions) =>
+        (validateWorkflowSource(partialSource) |@|
+          validateInputs(partialSource) |@|
+          validateOptions(partialSource.workflowOptions) |@|
+          validateWorkflowType(partialSource, version) |@|
+          validateWorkflowTypeVersion(partialSource, version)) map {
+          case (wfSource, wfInputs, wfOptions, workflowType, workflowTypeVersion) =>
             wfInputs.map(inputsJson => WorkflowSourceFilesCollection(
               workflowSource = wfSource,
-              workflowType = partialSource.workflowType,
-              workflowTypeVersion = partialSource.workflowTypeVersion,
+              workflowType = workflowType,
+              workflowTypeVersion = workflowTypeVersion,
               inputsJson = inputsJson,
               workflowOptionsJson = wfOptions.asPrettyJson,
               labelsJson = partialSource.customLabels.getOrElse("{}"),
-              importsFile = partialSource.zippedImports))        }
+              importsFile = partialSource.zippedImports,
+              warnings = partialSource.warnings))
+        }
       case Invalid(err) => err.invalid
     }
   }
 
-  private def deprecationWarning(out: String, in: String): Unit = {
+  private def deprecationWarning(out: String, in: String): String = {
     val warning =
       s"""
          |The '$out' parameter name has been deprecated in favor of '$in'.
          |Support for '$out' will be removed from future versions of Cromwell.
          |Please switch to using '$in' in future submissions.
          """.stripMargin
     log.warn(warning)
+    warning
```

Comment:
just remove `warning` declaration

Reply:
Can't remove this afaik. `warning` is used twice, once logged server side on the previous line, then returned here.